tools/squidclient/squidclient.cc

```diff
@@ -555,13 +544,13 @@ main(int argc, char *argv[])
 
         /* Send the HTTP request */
         debugVerbose(2, "Sending HTTP request ... ");
-        bytesWritten = Transport::Write(msg, strlen(msg));
+        bytesWritten = Transport::Write(messageHeader.c_str(), messageHeader.length());
 
         if (bytesWritten < 0) {
             std::cerr << "ERROR: write" << std::endl;
             exit(EXIT_FAILURE);
-        } else if ((unsigned) bytesWritten != strlen(msg)) {
-            std::cerr << "ERROR: Cannot send request?: " << std::endl << msg << std::endl;
+        } else if ((unsigned) bytesWritten != messageHeader.length()) {
+            std::cerr << "ERROR: Cannot send request?: " << std::endl << messageHeader << std::endl;
```

Comment:
I would say `Failed to send the following request:\n`. There should be no doubt that there was a sending failure.

It would be nice to report the error, but doing so properly requires too many out-of-scope changes, and I am not asking for that.

Reply:
done. Though that requires a very loose definition of "send" which includes a number of errors that can occur on a many-layered network connection (eg tunneled TLS) after delivery of the actual message bytes get delivered.